### PR TITLE
SPLICE-1335 Upgrade to Spark 2.1.0

### PR DESCRIPTION
**Pull request by jleach4**
This upgrade will unblock Jebril's work with null ordering.

### CODE REVIEW

File: hbase_sql/src/main/java/com/splicemachine/serializer/SpliceKryoSerializerInstance.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 public class SpliceKryoSerializerInstance extends KryoSerializerInstance {
 
     public SpliceKryoSerializerInstance(KryoSerializer ks) {
-        super(ks);
+        super(ks,false);
```
Comment (dgomezferro) [inline hbase_sql/src/main/java/com/splicemachine/serializer/SpliceKryoSerializerInstance.java]: Why not use Unsafe? Does it fail or is it slower?

Comment (jfilali) [inline hbase_sql/src/main/java/com/splicemachine/serializer/SpliceKryoSerializerInstance.java]: @jleach4 : Please review this comment, so wen can proceed to the merge after that clear out.

Comment (jleach4) [inline hbase_sql/src/main/java/com/splicemachine/serializer/SpliceKryoSerializerInstance.java]: Sorry for the delay, it fails with an underflow.  I think it is caused by some optimization in not encoding but I did not have time to chase it down.

Comment (jleach4) [inline hbase_sql/src/main/java/com/splicemachine/serializer/SpliceKryoSerializerInstance.java]: Meant to say int encoding

**Pull request merged**